src/components/News.jsx

```diff
@@ -35,21 +45,25 @@ class News extends React.Component {
       curData: [],
     };
 
-    this.setData = this.setData.bind(this);
+    this.setTrendingData = this.setTrendingData.bind(this);
+    this.setMyData = this.setMyData.bind(this);
     this.handleClick = this.handleClick.bind(this);
   }
 
   componentDidMount() {
-    this.setData(this.props.coin);
+    this.setTrendingData(this.props.coin);
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.coin !== nextProps.coin) {
-      this.setData(nextProps.coin);
+    if (this.props.coin !== nextProps.coin && this.state.trendingSelected) {
+      this.setTrendingData(nextProps.coin);
+    }
+    if (!checkListEquality(this.props.list, nextProps.list)) {
```

Comment:
if props are made state variables inside this component, then setState will handle the deep Comparison for us? So can we make props as state variables here?

Reply:
I guess then we won't require componentWillRecieveProps too along with setTrendingData etc..?

Reply:
My understanding is that setting props equal to state is seen as a bad practice (anti-pattern) for react:
https://medium.com/@justintulk/react-anti-patterns-props-in-initial-state-28687846cc2e

You're supposed to use lifecycle functions like componentWillReceiveProps to monitor prop changes.